shell/platform/windows/client_wrapper/flutter_engine.cc

```diff
@@ -88,6 +88,18 @@ FlutterDesktopPluginRegistrarRef FlutterEngine::GetRegistrarForPlugin(
   return FlutterDesktopEngineGetPluginRegistrar(engine_, plugin_name.c_str());
 }
 
+void FlutterEngine::SetNextFrameCallback(const std::function<void()>& callback) {
```

Comment:
Since you're almost certainly going to be passing a lambda to this function, I suspect it's actually more efficient to pass by value. In the pass by value case, a temporary std::function is constructed from the lambda literal and since it's an xvalue, will be moved. You can then std::move the callback to next_frame_callback and you'll never make a copy.

If you pass by const reference, we're still building a temporary xvalue std::function at the call site, then passing a reference, but you're then making a copy on line 92 here, which makes it less efficient.

Reply:
![C++ noob](https://media2.giphy.com/media/3oz8xLlw6GHVfokaNW/giphy.gif?cid=790b76118e9ba874e92197512ced2ff1648d47d9e055c069&rid=giphy.gif&ct=g)

Reply:
Yeah. If the implementation has to move the value, the passing it via const reference would make it impossible to avoid a copy. Passing it via value would make still make it incumbent on the caller to move it in to avoid a copy. But if the caller does this, you'll have no copies just moves. Of course the caller now has to make sure they don't run into use-after-moves. So there's that. Lambda capture behavior is separate.

In this particular instance, since the callback is moved (or should be) further, I'd prefer to do this by value.

Reply:
Thanks for the detailed background info, I appreciate it! I switched const references to moves.